sympy/sets/sets.py

```diff
@@ -1968,6 +1968,14 @@ def __lt__(self, other):
             raise TypeError("Invalid comparison of set with %s" % func_name(other))
         return self.is_proper_subset(other)
 
+    def __eq__(self, other):
```

Comment:
This looks like it's part of a bigger problem relating to equality of rationals and floats. I would just XFAIL the test and open a new issue for it (and remove this `__eq__`). The problem is that `Rational(3, 10) == S(0.3)` but `hash(3, 10) != hash(S(0.3))`. You can see my blog post about this to see why this is bad: https://asmeurer.github.io/blog/posts/what-happens-when-you-mess-with-hashing-in-python/.